.github/PULL_REQUEST_TEMPLATE/module_update.md

```diff
@@ -0,0 +1,64 @@
+<!--
+This project's monthly turnover might well be beyond 1500 PRs. Since we are community driven,
+let's do tit for tat: submitters are kindly ask to pick an open pull request for review.
+
+Start with packages updates. They are relatively simple.
+Since you'll be required to do those very steps below, it'll be easy.
+
+Marvin needs_reviewer: https://github.com/NixOS/nixpkgs/pulls?q=is%3Apr+is%3Aopen+label%3Aneeds_reviewer+
+Marvin needs_merger:   https://github.com/NixOS/nixpkgs/pulls?q=is%3Apr+is%3Aopen+label%3Aneeds_merger+
+Reviewing guidelines:  https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
+-->
+
+##### Motivation for this change
+
+
+##### General review
+
+- [ ] changes are backward compatible
+- [ ] removed options are declared with `mkRemovedOptionModule`
+- [ ] changes that are not backward compatible are documented in release notes
+- [ ] module tests succeed on ARCHITECTURE
+- [ ] options types are appropriate
+- [ ] options description is set
+- [ ] options example is provided
+- [ ] fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
+
+##### Package build
+<!-- Tipp: pipe those commands through bash directly from your editor -->
+
+```
+uname -srm
+---
+lsb_release -a
+```
+
+<details>
+<summary>Logs from <code>nixpkgs-review rev HEAD</code></summary>
+
+```
+nixpkgs-review rev HEAD
+
+```
+</details>
+
+<!--
+substitute this command by it's own output when run within the nipkgs-review shell
+-->
+cat ./report.md
+
+<!--
+substitute this command by it's own output when run within the nipkgs-review shell
+and test "not ok" binaries manually
+-->
+for cmd in $(ls ./results/**/**/**); do if $($cmd --help > /dev/null); then echo "- [x] \`$cmd --help\`: ok"; else "- [ ] \`$cmd --help\`: not ok -- tested otherwise"; fi; done
```

Comment:
In this section it seems as if running `nixpkgs-review` is obligatory. This is not a good idea though if you want to file e.g. mass-rebuilds in a PR though.

Reply:
Can you help me understand those mass rebuilds? What they are?
Should they rather have a separate PR template?

Reply:
Nix is a functional package manager which also means that if you have changed inputs, the outputs change as well (i.e. if a `buildInput` changes, the affected derivation changes and has to be rebuilt).

For low-priority packages this isn't a problem, but if you update e.g. `openssl` or `glibc` (almost every single package depends on the latter for instance), a lot of packages have to be rebuilt. To make sure that folks using `master` to build their systems don't have to rebuild the world, those changes land on a different branch named `staging` first (see also https://github.com/NixOS/rfcs/blob/master/rfcs/0026-staging-workflow.md).

For those kind of packages it's basically impossible to run `nix-review`, so this shouldn't be listed as step to file a PR.

> Should they rather have a separate PR template?

Actually this is a good idea. We could add the following fields to such a template:

* Corresponding Hydra jobset (if any)
* Tested architectures (whether the contributor tested against x86/aarch64/etc).

Reply:
Am I understanding correctly this is only relevant for `package_update`, not for `module_update`?

Reply:
Yes.

Reply:
Ok thanks added a new template and some additional explication. Since mass rebuilds be done by core maintainers the checklist is less descriptive, assuming people already know what they  are doing.